webpack/assets/javascripts/foreman_tools.js

```diff
@@ -103,44 +103,29 @@ export function initTypeAheadSelect(input) {
 // handle table updates via turoblinks
 export function updateTable(element) {
   const uri = new URI(window.location.href);
-  let searchTerm;
-  let perPage;
-  let isSearchForm;
-  let pageNum;
-
   if (element !== undefined) {
-    isSearchForm = element.id === 'search-form';
-
-    pageNum = $('#cur_page_num').val();
-    if (pageNum !== undefined) {
-      uri.setSearch('page', pageNum);
+    const values = { page: '', search: '', per_page: '' };
+    if (['per_page', 'search-form'].indexOf(element.id) > -1) {
+      values.page = '1';
     }
 
-    if (isSearchForm || element.id === 'per_page') {
-      uri.setSearch('page', 1);
-    }
-
-    if (isSearchForm) {
-      searchTerm = $(element)
-        .find('.autocomplete-input')
-        .val();
-      if (searchTerm) {
-        uri.setSearch('search', searchTerm.trim());
-      }
-    }
-
-    perPage = $('#per_page').val();
-    if (
-      perPage !== undefined &&
-      $('#search-form')
-        .find('.autocomplete-input')
-        .val() === undefined
-    ) {
-      uri.removeSearch('search');
-    }
-    uri.setSearch('per_page', perPage);
+    values.search =
+      element.id === 'search-form'
+        ? (
+          $(element)
+            .find('.autocomplete-input')
+            .val() || values.search
+        ).trim()
+        : '';
+    values.page = $('#cur_page_num').val() || values.page;
+    values.per_page = $('#per_page').val() || values.per_page;
+    Object.keys(values).forEach(k => uri.setSearch(k, values[k]));
```

Comment:
I am not sure I understand. Having the `param1="value1"` in the URI parameters and setting it to undefined doesn't remove it from URI, it just sets the search in URI to `param1`.

Can you give an example or something because I would like to remove that line if it's possible.